parmed/charmm/parameters.py

```diff
@@ -330,7 +330,7 @@ def read_parameter_file(self, pfile, comments=None):
             if line.startswith('ANGLE') or line.startswith('THETA'):
                 section = 'ANGLES'
                 continue
```

Comment:
It's a long loop body (which is probably a sign that it should be refactored at some point). So that's safest. 

But yes, I think it's necessary here. Processing each section occurs in a separate set of if/else blocks than processing the headers, so when a header is processed it needs to skip the parameter parsing logic (since its on a header line, not a parameter line).